hyperspy/signal.py

```diff
@@ -3111,6 +3112,83 @@ def crop(self, axis, start=None, end=None, convert_units=False):
         if convert_units:
             self.axes_manager.convert_units(axis)
 
+    def interpolate_on_axis(self,
+                            new_axis,
+                            replace_axis_index,
```

Comment:
Yes, I was unsure how to name this parameter, because there are 2 axes and hence I didn't want to call it `axis`. I changed it to `index_in_axes_manager` now, because it suggests that this parameter specifies the axis that will be replaced.